qualtran/bloqs/chemistry/thc/prepare.py

```diff
@@ -361,124 +368,138 @@ def selection_registers(self) -> Tuple[Register, ...]:
+        )
+        return qroam
+
+    def add_qrom(self, bb: 'BloqBuilder', **soqs: 'SoquetT') -> Dict[str, 'SoquetT']:
+        qrom = self.build_qrom_bloq()
+        # The qroam_junk_regs won't be present initially when building the
+        # composite bloq as they're RIGHT registers.
+        qroam_out_soqs = bb.add_d(qrom, selection=soqs['s'])
+        out_soqs: Dict[str, 'SoquetT'] = {'s': qroam_out_soqs.pop('selection')}
+        # map output soqs to Prepare junk registers names
+        out_soqs |= {
+            reg.name: qroam_out_soqs.pop(f'target{i}_')
+            for (i, reg) in enumerate(self.qroam_target_registers)
+        }
+        out_soqs |= {
+            reg.name: qroam_out_soqs.pop(f'junk_target{i}_')
+            for (i, reg) in enumerate(self.qroam_extra_target_registers)
+        }
+        return soqs | out_soqs
+
+    def build_composite_bloq(self, bb: 'BloqBuilder', **soqs: 'SoquetT') -> Dict[str, 'SoquetT']:
         # 1. Prepare THC uniform superposition over mu, nu. succ flags success.
-        mu, nu, succ, nu_eq_mp1, rot = bb.add(
+        soqs['mu'], soqs['nu'], soqs['succ'], soqs['nu_eq_mp1'], soqs['rot'] = bb.add(
             UniformSuperpositionTHC(num_mu=self.num_mu, num_spin_orb=self.num_spin_orb),
-            mu=mu,
-            nu=nu,
-            succ=succ,
-            nu_eq_mp1=nu_eq_mp1,
-            rot=rot,
+            mu=soqs['mu'],
+            nu=soqs['nu'],
+            succ=soqs['succ'],
+            nu_eq_mp1=soqs['nu_eq_mp1'],
+            rot=soqs['rot'],
         )
         data_size = self.num_spin_orb // 2 + self.num_mu * (self.num_mu + 1) // 2
         log_mu = self.num_mu.bit_length()
         log_d = (data_size - 1).bit_length()
         # 2. Make contiguous register from mu and nu and store in register `s`.
-        mu, nu, s = bb.add(ToContiguousIndex(log_mu, log_d), mu=mu, nu=nu, s=s)
+        soqs['mu'], soqs['nu'], soqs['s'] = bb.add(
+            ToContiguousIndex(log_mu, log_d), mu=soqs['mu'], nu=soqs['nu'], s=soqs['s']
+        )
         # 3. Load alt / keep values
-        qroam = SelectSwapQROM.build_from_data(
-            *(self.theta, self.alt_theta, self.alt_mu, self.alt_nu, self.keep),
-            target_bitsizes=(1, 1, log_mu, log_mu, self.keep_bitsize),
-            use_dirty_ancilla=False,
+        soqs |= self.add_qrom(bb, **soqs)
+        soqs['sigma'] = bb.add(OnEach(self.keep_bitsize, Hadamard()), q=soqs['sigma'])
+        lte_gate = LessThanEqual(self.keep_bitsize, self.keep_bitsize)
+        soqs['keep'], soqs['sigma'], soqs['less_than'] = bb.add(
+            lte_gate, x=soqs['keep'], y=soqs['sigma'], target=soqs['less_than']
         )
-        alt_mu, alt_nu = alt_mn
-        s, theta, alt_theta, alt_mu, alt_nu, keep = bb.add(
-            qroam,
-            selection=s,
-            target0_=theta,
-            target1_=alt_theta,
-            target2_=alt_mu,
-            target3_=alt_nu,
-            target4_=keep,
+        # TODO remove these
```

Comment:
Can you add some more context for this TODO ?

Reply:
Oh, I think this is done. Removed.